src/pages/organize/[orgId]/campaigns/[campId]/calendar/tasks/[taskId]/assignees.tsx

```diff
@@ -45,33 +50,84 @@ export const getServerSideProps : GetServerSideProps = scaffold(async (ctx) => {
     };
 }, scaffoldOptions);
 
+enum QUERY_STATUS {
+    EDITABLE='editable',
+    PUBLISHED='published',
+    PUBLISHING='publishing',
+}
+
 const TaskAssigneesPage: PageWithLayout = () => {
     const { taskId, orgId } = useRouter().query;
     const taskQuery = useQuery(['task', orgId, taskId], getTask(orgId as string, taskId as string));
+    const assignmentsQuery = useQuery(['assignments', orgId, taskId], getTaskAssignments(
+        orgId as string, taskId as string,
+    ));
     const task = taskQuery?.data;
-
+    const assignments = assignmentsQuery?.data;
     const query = task?.target;
 
     const [dialogOpen, setDialogOpen] = useState(false);
 
     const handleDialogClose = () => setDialogOpen(false);
 
+    const taskStatus = task ? getTaskStatus(task) : undefined;
+
+    const getQueryStatus = () => {
+        let queryStatus = QUERY_STATUS.PUBLISHED;
+        if (taskStatus === TASK_STATUS.DRAFT || taskStatus === TASK_STATUS.SCHEDULED) {
+            queryStatus = QUERY_STATUS.EDITABLE;
+        }
+        // we don't want 'publishing' state to appear on page load while the data is being fetched
+        else if (assignments && !assignments.length) {
+            queryStatus = QUERY_STATUS.PUBLISHING;
+        }
+        return queryStatus;
+    };
+
+    const queryStatus = getQueryStatus();
```

Comment:
Great solution to the complex logic! Putting it in a function hides it so that we don't have to think about it when reading the rendering logic. I would move the function outside of the component though, to get it completely out of the way.

Reply:
- but, the logic is also dependent on what is returned from the query that gets the `assignedTasks`, how does this work when moving outside of component?

Reply:
I put it in a function just because i don't like using `let` unless it's within scope of its own function, I don't know if that is a smart thing to do or doesn't matter here.

Reply:
> I put it in a function just because i don't like using `let` unless it's within scope of its own function, I don't know if that is a smart thing to do or doesn't matter here (that's why I used the double ternary at first)

I guessed as much, and using `const` is absolutely superior, which is why I think it's a great idea to move it to a function.

If you need stuff in the function (like the `queryStatus` and `taskStatus` etc), just pass them as arguments.